ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -841,20 +836,22 @@ private void readFromFile(File fileToRead, Journal journal) throws StoreExceptio
           index.computeIfAbsent(key, k -> new ConcurrentSkipListSet<>()).add(blobValue);
           logger.trace("IndexSegment : {} putting key {} in index offset {} size {}", indexFile.getAbsolutePath(), key,
               blobValue.getOffset(), blobValue.getSize());
-          // regenerate the bloom filter for in memory indexes
-          if (!isPresent) {
-            bloomFilter.add(ByteBuffer.wrap(key.toBytes()));
-          }
-          // add to the journal
-          long oMsgOff = blobValue.getOriginalMessageOffset();
-          if (oMsgOff != IndexValue.UNKNOWN_ORIGINAL_MESSAGE_OFFSET && offsetInLogSegment != oMsgOff
-              && oMsgOff >= startOffset.getOffset()
-              && journal.getKeyAtOffset(new Offset(startOffset.getName(), oMsgOff)) == null) {
-            // we add an entry for the original message offset if it is within the same index segment and
-            // an entry is not already in the journal
-            journal.addEntry(new Offset(startOffset.getName(), blobValue.getOriginalMessageOffset()), key);
+          if (!sealed.get()) {
```

Comment:
Just so I understand this right: if index is to be kept entirely in memory, we do not load the bloom filter associated with the sealed segments, but we still create and maintain them for active segments, and persist them when the segment is persisted. Is that correct? That seems like the right thing to do since the config may change in the future.

Reply:
Yup - sealing + mmap + persisting bloom filter are all decoupled. So even if an `IndexSegment` is meant to be in memory, it will still persist the index segment file + bloom filter.